docs/design/eth-filter.md

```diff
@@ -0,0 +1,133 @@
+
+### Initiating a filtration
+
+#### Request
+
+```javascript
+{"jsonrpc":"2.0","id": 1, "method": "eth_newFilter", "params": [fromBlock, toBlock, address, topics]}
+```
+
+#### Response
+
+```javascript
+{"jsonrpc":"2.0","id": 1, "result": FILTER_ID}
+```
+
+### Getting Filter Changes
+
+     The `getFilterLogs` method returns logs that match a specified topic filter and are included in newly added blocks.
+
+#### Parameters
+
+An existing FILTER_ID
+
+#### Request
+
+```javascript
+{"jsonrpc":"2.0","id": 1, "method": "eth_getFilterLogs", "params": ["0x123"]}
+```
+
+#### Response
+
+```javascript
+{
+    "jsonrpc": "2.0",
+    "id": 73,
+    "result": [{
+        "address": "0xb5a5f22694352c15b00323844ad545abb2b11028",
+        "blockHash": "0x99e8663c7b6d8bba3c7627a17d774238eae3e793dee30008debb2699666657de",
+        "blockNumber": "0x5d12ab",
+        "data": "0x0000000000000000000000000000000000000000000000a247d7a2955b61d000",
+        "logIndex": "0x0",
+        "removed": false,
+        "topics": ["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef", "0x000000000000000000000000bdc0afe57b8e9468aa95396da2ab2063e595f37e", "0x0000000000000000000000007503e090dc2b64a88f034fb45e247cbd82b8741e"],
+        "transactionHash": "0xa74c2432c9cf7dbb875a385a2411fd8f13ca9ec12216864b1a1ead3c99de99cd",
+        "transactionIndex": "0x3"
+    }]
+}
+```
+
+## Error Codes
+
+| Error Codes |   Error message   |                              Solution                               |
+| :---------: | :---------------: | :-----------------------------------------------------------------: |
+|    32000    | Filter not found. | Occurs when user attempts receive logs using non-existing filterID. |
+
+## Rate Limits
+
+1. All filters expire after 5 minutes of inactivity (no queries).
+
+## Metric Capturing
+
+Capture metrics for the following:
+
+1. Log every call to all filter API method, as a total amount and broken down by IP.
+2. The duration of active filters.
+3. The total amount of requests to the mirror node per filter.
+4. The amount of requests to the mirror node per filter that have returned non-null data.
+
+## Tests
+
+The following test cases should be covered but additional tests would be welcome.
+
+1. Overall functionality of creating and uninstalling new filters.
+2. Receiving requested information depending on filterID using `eth_getFilterChanges` and `eth_getFilterLogs`.
+3. Deleting of filter due to inactivity.
+4. E2E test using popular libraries (`ethers.js WebSocketProvider`).
+
+## Non-Functional Requirements
+
+Users should be required to renew their filters in the case of an inactivity. If for some reason the relay is restarted it will lose all cached filters.
+
+## Deployment
+
+Filter API will run alongside the already available HTTP server.
+
+## Answered Questions
+
+1. How will we implement the `getFilterChanges` and `getFilterLogs` flow?
```

Comment:
I'm not sure 1 & 2 are fully answered.
As part of the design doc I'd like to get a better sense of what classes are involved, both existing and new ones.
These changes should be called out in summary

Reply:
Yes, you are right. I've reworked most of the doc.

Reply:
I don't think you've answered the flow or interfaces much.
This isn't new logic i guess since it's similar to `eth_getLogs`